mono/utils/mono-state.c

```diff
@@ -927,6 +927,34 @@ mono_native_state_add_memory (MonoStateWriter *writer)
 	mono_state_writer_printf(writer, "},\n");
 }
 
+static void
+mono_native_state_add_process_map (MonoStateWriter *writer)
+{
+#if defined(__linux__) && !defined(HOST_ANDROID)
```

Comment:
why can't we do this on `HOST_ANDROID`?

Reply:
Does android have /proc? I'm not sure why it was requested to be removed from their dumps initially. I think it was just unexpected. 

Also should have pushed to a non-PR branch, but FYI I'm currently debugging this. Sorry if you reviewed unready code. Should be done soon, just ironing out some C problems. `read` is an annoying syscall sometimes.